### ISSUE TITLE
Add support for Shell

### ISSUE THREAD

**Issue by sigzero** [kkoomen/vim-doge#48]
https://google.github.io/styleguide/shell.xml#Comments

This could work for all types (BASH, SH, KSH, etc.).

**Comment by cansurmeli** [comment]
Hey.

Yes, the fact that bash functions doesn't have arguments is one of the downsides of bash scripting for me. It's irritating and I get your point regarding not providing `vim-doge` support but yet again, doesn't it make sense to comment our bash functions and at least provide the input/output inside the documentation section.

I think the resource provided by @sigzero has a good example:

```
Function Comments
link ▽ Any function that is not both obvious and short must be commented. Any function in a library must be commented regardless of length or complexity.

It should be possible for someone else to learn how to use your program or to use a function in your library by reading the comments (and self-help, if provided) without reading the code.

All function comments should contain:

    Description of the function
    Global variables used and modified
    Arguments taken
    Returned values other than the default exit status of the last command run

Example:

#!/bin/bash
#
# Perform hot backups of Oracle databases.

export PATH='/usr/xpg4/bin:/usr/bin:/opt/csw/bin:/opt/goog/bin'

#######################################
# Cleanup files from the backup dir
# Globals:
#   BACKUP_DIR
#   ORACLE_SID
# Arguments:
#   None
# Returns:
#   None
#######################################
cleanup() {
  ...
}
```

Yes, yet again, the `cleanup()` can take arguments but at least the commenting above makes it a bit more visible.

This would especially be useful when dealing with a large code base of bash scripts. I've such one(maybe not enormeous) but still big and separated through different files and commenting would make the code much more readable.

Thanks.

**Comment by kkoomen** [comment]
@cansurmeli I will re-consider implementing this if you can find me a proper styleguide that could cover this, or you can be the first one to create a proper bash styleguide and then I'd be honored to immediately implement it. I need a name for the styleguide for my plugin, that's why I'm asking as well. If you want to create your own styleguide (since I can't recall that I've find a bash styleguide at all) I volunteer in helping out as well.

Bash implementation will be super quick, because there are only two possible ways to declare a function as far as I know, which are: `function <name> { ... }` or `<name>() { ... }`. Do you perhaps know if a bash function using the `<name>() { ... }` notation can expect anything inside the parenthesis? I've _never_ seen this though.

I do agree with the arguments and the return type, these can be added easily. Can you elaborate more on the globals that should be defined? I write a lot of bash scripts as well, but the idea to me is not clear. Are you talking about the globals it is defining? Or about the globals you're modifying within the function? or something else?

**Comment by kkoomen** [comment]
Another thing: Google displays a very random amount of `#` characters on the top and bottom of the comment. I suggest we use an exact amount of 80 like doxygen for C as well.

**Comment by kkoomen** [comment]
Also, how does Google display arguments for a function?

For example, I accept two arguments, should I write something like this?

```sh

#######################################
# Cleanup files from the backup dir
# Globals:
#   BACKUP_DIR
#   ORACLE_SID
# Arguments:
#   $1: Lorem ipsum dor sit amet.
#   $2: Another comment.
# Returns:
#   None
#######################################
```

**Comment by sigzero** [comment]
That's probably how I would do it.

**Comment by kkoomen** [comment]
Agree. That'd be logical for me to use that notation.

This is what I have locally already as generated results:

```
################################################################################
# [TODO:description]
# Globals:
#   [TODO:var-name]
# Arguments:
#   $1: [TODO:description]
# Returns:
#   [TODO:description]
################################################################################
function test {
  print "foobar"
}

################################################################################
# [TODO:description]
# Globals:
#   [TODO:var-name]
# Arguments:
#   $1: [TODO:description]
# Returns:
#   [TODO:description]
################################################################################
test() {
  print "foobar"
}
```

If both of you @sigzero and @cansurmeli agree with this then I'll write tests and submit this right away.

**Comment by sigzero** [comment]
That works for me.

**Comment by kkoomen** [comment]
This feature has been merged and released in [v1.16.0](https://github.com/kkoomen/vim-doge/tree/v1.16.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.

**Comment by cansurmeli** [comment]
I think this is a great addition. It's well worth it. Thank you for your efforts and expedited act.
